c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandler.java

```diff
@@ -0,0 +1,196 @@
+        } catch (Exception e) {
+            LOG.error("Unexpected error happened", e);
+            operationState = operationState(NOT_APPLIED, UNABLE_TO_CREATE_BUNDLE);
+        } finally {
+            ofNullable(contentFilteredFilePaths).ifPresent(this::cleanup);
+        }
+
+        LOG.debug("Returning operation ack with state {} and details {}", operationState.getState(), operationState.getDetails());
+        return operationAck(operation, operationState);
+    }
+
+    private C2OperationAck operationAck(C2Operation operation, C2OperationState state) {
+        C2OperationAck operationAck = new C2OperationAck();
+        operationAck.setOperationId(ofNullable(operation.getIdentifier()).orElse(EMPTY));
+        operationAck.setOperationState(state);
+        return operationAck;
+    }
+
+    private C2OperationState operationState(OperationState operationState, String details) {
+        C2OperationState state = new C2OperationState();
+        state.setState(operationState);
+        state.setDetails(details);
+        return state;
+    }
+
+    private List<Path> filterContent(List<Path> bundleFilePaths) {
+        List<Path> contentFilteredFilePaths = new ArrayList<>();
+        for (Path path : bundleFilePaths) {
+            String fileName = path.getFileName().toString();
+            try (Stream<String> fileStream = lines(path)) {
+                Path tempDirectory = createTempDirectory(null);
+                Path tempFile = Paths.get(tempDirectory.toAbsolutePath().toString(), fileName);
+                write(tempFile, (Iterable<String>) fileStream.filter(contentFilter)::iterator);
+                contentFilteredFilePaths.add(tempFile);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return contentFilteredFilePaths;
+    }
+
+    private Optional<byte[]> createDebugBundle(List<Path> filePaths) {
+        ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+        try (GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(byteOutputStream);
+             TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(gzipCompressorOutputStream)) {
+            for (Path filePath : filePaths) {
+                TarArchiveEntry tarArchiveEntry = new TarArchiveEntry(filePath.toFile(), filePath.getFileName().toString());
+                tarOutputStream.putArchiveEntry(tarArchiveEntry);
+                copy(filePath, tarOutputStream);
+                tarOutputStream.closeArchiveEntry();
+            }
+            tarOutputStream.finish();
+        } catch (Exception e) {
+            LOG.error("Error during create compressed bundle", e);
+            return empty();
+        } finally {
+            closeQuietly(byteOutputStream);
+        }
+        return Optional.of(byteOutputStream).map(ByteArrayOutputStream::toByteArray);
+    }
+
+    private void cleanup(List<Path> paths) {
```

Comment:
It doesn't change the world I just looked it up because it was interesting. If you have the base dir you can use the file walker approach which will iterate over everything in the dir
```
try (Stream<Path> walk = Files.walk(baseDirectory)) {
                walk.map(Path::toFile).forEach(File::delete);
            }
```
https://stackoverflow.com/questions/35988192/java-nio-most-concise-recursive-directory-delete

Your approach is slightly different as you have the list of files to delete but in the end you want to delete the whole (hopefully empty) directory in the end.
Regardless I'm fine with the current approach as well.

Reply:
Changed the approach

Reply:
Looks good, thanks